finality-aleph/src/sync/substrate/verification/cache.rs

```diff
@@ -236,12 +258,14 @@ mod tests {
     fn setup_test(max_session_n: u32, finalized_number: &'_ Cell<u32>) -> TestVerifierCache<'_> {
         let finalization_info = MockFinalizationInfo { finalized_number };
         let authority_provider = MockAuthorityProvider::new(max_session_n);
+        let genesis_header = MockHeader::random_parentless(0);
 
         VerifierCache::new(
             SessionPeriod(SESSION_PERIOD),
             finalization_info,
             authority_provider,
             CACHE_SIZE,
+            genesis_header,
```

Comment:
Nope, it's annoyingly hard to make one, because this problem only occurs in the substrate implementation of our traits (the internal code doesn't know one kind of justification is special), and to test the substrate implementation in this context we would have to produce a `Header` which is annoyingly hard to do for testing purposes. :/